src/modules/endpoint/fetchers/extras.js

```diff
@@ -0,0 +1,30 @@
+// @flow
+
+import { apiUrl } from '../constants'
+import ExtraModel from '../models/ExtraModel'
+import type { Dispatch } from 'redux-first-router/dist/flow-types'
+import { saveExtras } from '../actions/fetcher'
+
+type Params = {
+  city: string,
+  language: string
+}
+
+const urlMapper = (params: Params): string => `${apiUrl}/${params.city}/${params.language}/wp-json/extensions/v3/extras`
+
+const fetcher = (dispatch: Dispatch, params: Params): Promise<Array<ExtraModel>> =>
+  fetch(urlMapper(params))
+    .then(result => result.json())
+    .then(json => json
+      .map(extra => new ExtraModel({
+        alias: extra.alias,
+        name: extra.name,
+        path: extra.url,
+        thumbnail: extra.thumbnail
+      })))
+    .then(extras => {
+      dispatch(saveExtras(extras))
```

Comment:
I think this is more complicated as it needs to be. It's absolutly not clear which action triggers which other actions and results in which store diff.

Reply:
Suggestions? There is only one dispatch here and this is saving the extras to the store